#Unix Assignment 1 Readme fileYour UNIX assignment will consist of two components:#Start of Project##Inspection of Data Filesfile command for file time, non aski characters? add this and awk, look at the header, might not have collumns in header as in data* First Start with inspection of fang_et_al_genotypes.txt   * Logins and start with fang_et_al_genotypes.txt   * use the wc command fang_et_al_genotypes.txt```	 2783  2744038 11051939 fang_et_al_genotypes.txt   #this was the output of wc    		command. ```  * above means that this file has 2783 lines, 2744038 words, and 11051939 bytes  * next use the du -h command to see more human readable   ```*  $ du -h fang_et_al_genotypes.txt 11M     fang_et_al_genotypes.txt```  * from this we can easily gather the file has 11megabytes of information   * next we will use the file command to make sure that all characters are ASCII  *  ```$ file fang_et_al_genotypes.txtfang_et_al_genotypes.txt: ASCII text, with very long lines``` * this tells us that all of our characters are ASCII but the file has very long lines! * next I used the cut and head commands to get a better idea of what the file looked like, using head -n 3 still gave me a massive amount of info so use this: * ```$ cut -f 1-10 fang_et_al_genotypes.txt | head``` * output: tells me that the snp data does not start till the second row and fourth column. the first row starts out as Sample_ID, JG_OTU, Group, and then the snp info starts * Next use the awk command to find the number of columns:	```$ awk -F "\t" '{print NF; exit}' fang_et_al_genotypes.txt  ``` *this line of code tells me there is 986 columns in the file!... * Now lets inspect the second file ```snp_position.txt``` * using the same commands from above I will gather the information *```wc```   984 lines (less lines), 13198 letters (less letters), 83747 characters (less characters) * ```du -h``` 84K, so this file has 84kb *```file```snp_position.txt: ASCII text, with CRLF line terminators      this means that it is in ASCII but with CRLF line terminators (we will probably need to address this in joining of files) * next I will look at the file to try to get an idea of what is in the columns, best to use a cut command.  ``` cut -f 1-6 snp_position.txt``` I am told that column 1 is SNP_ID, 2 is cdv_marker_id, 3 is chromosome, 4 is position, and five is alt_Position, 6 is mult_positions, so on, SNP data starts the second row after the header.  *now lets check the number of colummns using awk, I should have done this before cut. It tells me that there are 15 columns is this file. ```$ awk -F "\t" '{print NF; exit}' snp_position.txt15```## Data Processing### It startsCode progression for markdown readme* Started with getting the teosinte vs maize genotypes sepperated into their own file```	$ grep -w "ZMMIL" fang_et_al_genotypes.txt > maize_genotypes.txt```You have to do this and change the group (this case ZMMIL) to another maize group to grep but append using >>```	$ grep -w "ZMMLR" fang_et_al_genotypes.txt >> maize_genotypes.txt```Must also do this for “ZMMMR” for maize .Repeat above steps but for the teosinte groups “ZMPBA, ZMPIL, and ZMPJA”Now you have files that will contain all info of the genotypes files separated by groups Maize and Teosinte.* I also “grepped” in the header, you don’t need to do this but I liked it to keep things organized```	grep -w "Sample_ID" fang_et_al_genotypes.txt >> transposed_maize_genotypes.txt```grep –w recognized “Sample_ID” from the header and appended the whole line to the file above* next I transposed both my maize and teoxinte genotype files so that they would have a column that would match with the snp document (now all first column)```	$ awk -f transpose.awk teosinte_genotypes.txt > transposed_teosinte_genotypes.txt	$ awk -f transpose.awk maize_genotypes.txt > transposed_maize_genotypes.txt```* Next I sorted all three files, snps, maize and teosinte genotypes so that they sorted all based off the first column  (the matching one with all three)```	$ sort -k1,1 transposed_maize_genotypes.txt > sorted_transposed_maize_genotypes.txt	$ sort -k1,1 snp_position.txt > sorted_snp_position.txt	$ sort -k1,1 transposed_teosinte_genotypes.txt > sorted_transposed_teosinte_genotypes.txt```* Next I joined my genotype files with my snp file, I got the proper join code from slack```	$ join -t $'\t' -1 1 -2 1 sorted_snp_position.txt sorted_transposed_maize_genotypes.txt > joined_sorted_transposed_maize_genotypes.txt	$ join -t $'\t' -1 1 -2 1 sorted_snp_position.txt sorted_transposed_teosinte_genotypes.txt > joined_sorted_transposed_teosinte_genotypes.txt```* After my two files were joined I cut out the columns the assignment calls for, 1st “SNP_ID”, 2nd “Chromosome”, 3rd “Position”, and  subsequent columns are genotype are from either maize or teosinte individuals. I took the output of that and made a new full possessed genotype + SNP folders. ```	$ cut -f 1,3,4,16- joined_sorted_transposed_maize_genotypes.txt > cut_joined_maize.txt	$ cut -f 1,3,4,16- joined_sorted_transposed_teosinte_genotypes.txt > cut_joined_teosinte.txt```* For the first 40 desired files from the assignment for maize and teosinte I needed to separate each chromosome out (10 total). So I utilized a for loop that another student posted on slack and modified it to work with my data by changing the names of files.```	$ for i in {1..10}; do awk '$2=='$i'' cut_joined_teosinte.txt > chr"$i"_teosinte.txt; done	$ for i in {1..10}; do awk '$2=='$i'' cut_joined_maize.txt > chr"$i"_maize.txt; doneowner@DESKTOP-SIR8KL0 MINGW64 ~/Desktop/BCB_546X/BCB546X-Fall2017/Unix_Assignment_1_JMA/UNIX_Assignment (master)$ lschr1_maize.txt      chr4_teosinte.txt  chr9_maize.txt                                   sorted_snp_position.txtchr1_teosinte.txt   chr5_maize.txt     chr9_teosinte.txt                                sorted_transposed_maize_genotypes.txtchr10_maize.txt     chr5_teosinte.txt  cut_joined_maize.txt                             sorted_transposed_teosinte_genotypes.txtchr10_teosinte.txt  chr6_maize.txt     cut_joined_teosinte.txt                          teosinte_genotypes.txtchr2_maize.txt      chr6_teosinte.txt  fang_et_al_genotypes.txt                         transpose.awkchr2_teosinte.txt   chr7_maize.txt     joined_sorted_transposed_maize_genotypes.txt     transposed_maize_genotypes.txtchr3_maize.txt      chr7_teosinte.txt  joined_sorted_transposed_teosinte_genotypes.txt  transposed_teosinte_genotypes.txtchr3_teosinte.txt   chr8_maize.txt     maize_genotypes.txt                              UNIX_Assignment.mdchr4_maize.txt      chr8_teosinte.txt  snp_position.txt                                 UNIX_Assignment.pdf``` as shown by this list command, this for loop created a file chr(ranges 1-10) for chromosome of that genotype and placed each       chromosome into their respectfully named file. I.E Chromosome 1 from the maize genotype file was extracted and placed into a  new file called chr1_maize.txt8. Now that I have each chromosome in its own respective file I began to process the files with SNPs ordered based on increasing position values, the missing data comes default as encoded by “?” so I did not need to change.```	$ sort -k3 -n chr1_teosinte.txt > chr1_increasingposition_teosinte.txt	$ sort -k3 -n chr1_maize.txt > chr1_increasingposition_maize.txt```So ^here we are sorting based of the 3rd column (position) of each and every chromosome file 1 through 10 for both maize and teosinte. I used the same code just changed the chromosome number for each run of the command. Default sort here will be from increasing position, no need to specify. This will create 20 files (10 each for maize and teosinte) * Next I began to process files with SNPs ordered based on decreasing position values.```	$ sed 's/?/-/g' chr1_teosinte.txt | sort -k3 -nr > chr1_decreasingposition_teosinte.txt	$ sed 's/?/-/g' chr1_maize.txt | sort -k3 -nr > chr1_decreasingposition_maize.txt```But before I can sort I needed to run the command “sed” to match for “?” symbol and replace it with “-“ because of what they specified they wanted for this file. I needed to use “sed” global option “g’” to indicate to change every “?” in the line to “-“, “sed” default will only change the first. After this was done I could then pipe the output of that through sort based off the 3rd column, but I needed to indicate “-r” so that out put would be in-reverse, instead of increasing position values, this makes the output decreasing values. . I used the same code just changed the chromosome number for each run of the command.. This will create 20 files (10 each for maize and teosinte)* For the 1 file for maize and teosinte that has all SNPs with unknown position in the genome, I used the “grep –w” command to search for the word “unknown”, this is how these are listed in the data. The option w “-w” will match the exact entire word in a line and will grab that entire line to standard out for each instance. I just simply then redirected standard out to a new file, 42/44 files of the assignment done```	$ grep -w "unknown" cut_joined_maize.txt > SNPs_with_unkown_position_maize.txt	$ grep -w "unknown" cut_joined_teosinte.txt > SNPs_with_unkown_position_teosinte.txt```* Lastly I needed to make 1 file for both maize and teosinte that has all SNPs with multiple positions in the genome. I used the “grep –w” command to search for the word “multiple”, this is how these are listed in the data. The option w “-w” will match the exact entire word in a line and will grab that entire line to standard out for each instance. 44/44 files of the assignment done```	$ grep -w "multiple" cut_joined_teosinte.txt > SNPs_with_multiple_positions_teosinte.txt	$ grep -w "multiple" cut_joined_maize.txt > SNPs_with_multiple_positions_maize.txt```# Notes:Between each steps of the analysis I would run check commands to see if my code did what I wanted it to. Between each step I would run variations awk, cut, file, head, and tail to view my output files to see that the change was correctly done.